src/audio/codec_adapter/codec/cadence.c

```diff
@@ -139,6 +139,48 @@ int cadence_codec_init(struct comp_dev *dev)
 	return ret;
 }
 
```

Comment:
I have mixed feelings about this. The logic should be this:
- set parameters from config (1)
- update/set some parameters from stream (2)

If the parameter is not set at (1) this doesn't mean we are not allowed to update it in (2).

So, I would suggesting introducing a new function:
```

static int init_params(struct comp_dev *dev)
{
      struct ca_config *ca_config = <>;

      switch (ca_config->id) {
      case CADENCE_AAC_ID:
                 ca_config->sample_rate.id = XA_AAC_SAMPLE_RATE_ID;
    default:
          break;
    case CADENCE_AAC_ID:
                ca_config->sample_rate.id = XA_MP3_SAMPLE_RATE_ID;
     default:
       break;
}
```

And update the ca_config structure like this:

```
+struct ca_param {
+       uint32_t id;
+       uint32_t value;
+}
+
 /** codec adapter setup config parameters */
 struct ca_config {
        uint32_t codec_id;
        uint32_t reserved;
-       uint32_t sample_rate;
-       uint32_t sample_width;
-       uint32_t channels;
+       struct ca_param sample_rate;
+       struct ca_param sample_width;
+       struct ca_param channels;
 };
```

Reply:
Thanks for the proposed solution. Before this line
`switch (ca_config->id) {
      case CADENCE_AAC_ID:`
we first need to check `codec_id` and then use apropriate parameters id like `CADENCE_AAC_ID` in your example.

Reply:
Actually if ca_config->id is not valid it will go on default usecase and then we print an error and return an error.